.github/workflows/ansible-test.yml

```
@@ -58,8 +58,39 @@ jobs:
         with:
           fail_ci_if_error: false
 
+  integration_matrix:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ansible_collections/community/mongodb
+    outputs:
+      matrix: ${{ steps.json-list.outputs.modules-to-test }}
+
+    steps:
+
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          path: ansible_collections/community/mongodb
+
+      - name: Generate module list
+        run: |
+          find tests/integration/targets -type d -maxdepth 1 -not -name 'setup*' | cut -d '/' -f4 | sort | awk NF > int_test_modules.txt
+
+      - name: Generate json doc from module list
+        run: |
+          cat int_test_modules.txt | jq -sRc 'split("\n")' > int_test_modules.json
```

```suggestion
          cat int_test_modules.txt | jq -sRc 'split("\n") | map(select(length > 0))' > int_test_modules.json
```
This should remove the empty string (the purpose of that perl task).

---

Got mised up here. We actually need to remove the final newline in the int_test_modules.txt
 file. Get a syntax error with the map command so perhaps it's not supported in the Ubuntu VM the CI runs in.

---

Looks like you added the map to the wrong task. This goes in the `jq` task (in "Generate json doc from module list" instead of "Generate module list").

```
$ cat int_test_modules.txt
sdf
qwer
$ cat int_test_modules.txt | jq -sRc 'split("\n")'
["sdf","qwer",""]
$ cat int_test_modules.txt | jq -sRc 'split("\n") | map(select(length >0))'
["sdf","qwer"]
$ jq --version
jq-1.6
```

---

You can also combine both "Generate module list" and "Generate json doc from module list" like this:
```yaml
      - name: Generate module list
        run: |
          find tests/integration/targets -type d -maxdepth 1 -not -name 'setup*' | cut -d '/' -f4 | sort | awk NF | jq -sRc 'split("\n") | map(select(length > 0))' > int_test_modules.json
```
Note that there is a `|` inside the jq script.

---

Ah yes, I didn't spot that.